docs/migration-guide.md

```diff
@@ -80,29 +103,235 @@ val MIGRATION_HREF = object : Migration(1, 2) {
+
+As it implements `InputListener`, you can attach it to any `OverflowableNavigator`.
+
+```kotlin
+navigator.addInputListener(
+    DirectionalNavigationAdapter(
+        navigator,
+        animatedTransition = true
+    )
+)
+```
+
+The `DirectionalNavigationAdapter` provides plenty of customization options. Please refer to its API for more details.
+
+#### Tap and drag events
+
+The `onTap` and `onDrag` events of `VisualNavigator.Listener` have been deprecated. You can now use multiple implementations of `InputListener`. The order is important when events are consumed.
+
+```kotlin
+navigator.addInputListener(DirectionalNavigationAdapter(navigator))
+
+navigator.addInputListener(object : InputListener {
+    override fun onTap(event: TapEvent): Boolean {
+        toggleUi()
+        return true
+    }
+})
+```
+
+### LCP
+
+#### Creating an `LcpService`
+
+The `LcpService` now requires an instance of `AssetRetriever` and `DownloadManager` during construction. To get the same behavior as before, you can use a `ForegroundDownloadManager`. If you want to support downloads in the background instead, take a look at `AndroidDownloadManager`.
+
+```kotlin
+val lcpService = LcpService(
+    context,
+    assetRetriever = assetRetriever,
+    downloadManager = ForegroundDownloadManager(
+        httpClient = httpClient,
+        downloadsDirectory = File(context.cacheDir, "lcp")
+    )
+)
+```
+
+#### Downloading an LCP protected publication from a license
+
+`LcpService.acquirePublication()` is deprecated in favor of `LcpService.publicationRetriever()`, which provides greater flexibility thanks to the `DownloadManager`.
+
+```kotlin
+// 1. Open an `Asset` from a `File`.
+val asset = assetRetriever.retrieve(file)
+    .getOrElse { /* Failed to open the file or sniff its format */ }
+    
+// 2. Verify that it is an LCP License Document.
+if (asset is ResourceAsset && asset.format.conformsTo(LcpLicenseSpecification)) {
+    // 3. Parse the LCP License Document from its JSON representation.
+    val license = lcplAsset.resource.read()
+        .getOrElse { /* Failed to read the content of the LCPL asset */ }
+        .let { LicenseDocument.fromBytes(it) }
+        .getOrElse { /* Failed to parse a valid LCP License Document from the the raw bytes */ }
+
+    // 4. Download the publication using the `LcpPublicationRetriever`.
+    // The returned `requestId` can be used to cancel an on-going download, or to resume a download
+    // with `LcpPublicationRetriever.register()`, if it was downloaded in the background.
+    val requestId = lcpService.publicationRetriever()
+        .retrieve(license, listener = object : LcpPublicationRetriever.Listener {
+            override fun onAcquisitionCompleted(
+                requestId: LcpPublicationRetriever.RequestId,
```

Comment:
It should be cleared that the callbacks can be called after an app restart.

Reply:
I briefly mentioned it in the comment above, but I think that should be part of the download manager user guide and/or the LCP one (I intend to write one soon). For the migration, I expect people will begin with the `ForegroundDownloadManager`.